end-to-end-tests/src/test/java/com/amazonaws/blox/AbstractEndToEndTest.java

```diff
@@ -31,7 +31,10 @@
 
   @Before
   public void setupBase() {
-    final String bloxEndpoint = System.getProperty("blox.tests.apiUrl");
+    String bloxEndpoint = System.getProperty("blox.tests.apiUrl");
+    if (bloxEndpoint == null || bloxEndpoint.isEmpty()) {
```

Comment:
You could use the two-argument form of `System.getProperty(key, default)` to simplify things here:
```
System.getProperty("blox.tests.apiUrl", System.getenv("BLOX_TESTS_APIURL"));
```

Reply:
good suggestions! will change